### PR TITLE
Add CONTRIBUTING.md file

### LINKED ISSUE #1017: CONTRIBUTING.md is not found in aws-sdk-java
I want to contribute aws-sdk-java, but I could not find _CONTRIBUTING.md_. Other sdks have it, like below.

* https://github.com/aws/aws-sdk-ruby/blob/master/CONTRIBUTING.md
* https://github.com/aws/aws-sdk-js/blob/master/CONTRIBUTING.md

I think it's better to put _CONTRIBUTING.md_ explicitly in this repository.

## reference

* https://github.com/aws/aws-sdk-java/issues/569

### PR DESCRIPTION
**Pull request by BigChief45**
Copied v2's `CONTRIBUTING.md` and replaced links.

Closes #1017

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+* [ ] Code is documented, especially public and user-facing constructs
+* [ ] Local run of `mvn install` succeeds
+* [ ] Git commit message is detailed and includes context behind the change
+* [ ] If the change is related to an existing Bug Report or Feature Request,
+  the issue number is referenced
+* [ ] A short description of the change added to
+  [CHANGELOG.md](./CHANGELOG.md). Adding a new entry can be accomplished by
+  running the `scripts/new-change` script and following the instructions.
+  Commit the new file created by the script in `.changes/next-release` with
+  your changes.
```
Comment (millems) [inline CONTRIBUTING.md]: This script isn't available for 1.11. The SDK team member that accepts your pull request will be able to add a changelog entry, though.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+to see if your change will require additional work to be accepted.
+
+#### Additional Pull Request Requirements
+##### Reactive Streams
+If the change includes implementations of the [Reactive Streams
+interfaces](https://github.com/reactive-streams/reactive-streams-jvm), the
+change must also contain verification tests using the [Reactive Streams
+Technology Compatibility
+Kit](https://github.com/reactive-streams/reactive-streams-jvm/tree/master/tck)
+to ensure specificiation compliance.
```
Comment (millems) [inline CONTRIBUTING.md]: Reactive streams aren't used in 1.11, so this can be removed.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+
+If your branch has more than one commit when it's approved, you will also be
+asked to [squash][git-rewriting-history] them into a single commit before it is
+merged in.
+
+## Additional Resources
+We maintain a [wiki][wiki] where information like design decisions, internal
+architecture, and style conventions are documented that you may find helpful
+when contributing to the SDK.
+
```
Comment (millems) [inline CONTRIBUTING.md]: Unfortunately this doesn't exist for 1.11.

File: CONTRIBUTING.md (hunk 1 above)
Comment (BigChief45) [reply CONTRIBUTING.md]: Understood. Will remove item from list then.

**Comment by BigChief45** [comment]
> If your branch has more than one commit when it's approved, you will also be
asked to [squash][git-rewriting-history] them into a single commit before it is
merged in.

@millems Does this still apply or do maintainers just do a squash merge of the PR?

**Comment by BigChief45** [comment]
@millems I have applied the requested changes. Let me know when this is ready so I can squash.

**Comment by millems** [comment]
@BigChief45 LGTM! Are you fine submitting these changes under the Apache 2.0 license?

**Comment by BigChief45** [comment]
@millems I have squashed all commits into one.

_I confirm that I am submitting these changes under the Apache 2.0 license._

**Comment by millems** [comment]
Thank you very much for your contribution!

**Pull request merged**